### PR TITLE
Keep a thread safe in-memory cache of generated diagrams with links

### PR DESCRIPTION
**Pull request by dirkgroot**
This should fix the error reported in #279.

### CODE REVIEW

File: src/main/kotlin/nl/avisi/structurizr/site/generatr/site/DiagramGenerator.kt [hunk 1]
```diff
@@ -34,22 +36,18 @@ fun generateDiagrams(workspace: Workspace, exportDir: File) {
         }
 }
 
-fun generateDiagramWithElementLinks(workspace: Workspace, view: View, url: String, exportDir: File): String {
-    val pumlDir = pumlDir(exportDir)
-    val svgDir = svgDir(exportDir)
-
+private val diagramCache = ConcurrentHashMap<String, String>()
+fun generateDiagramWithElementLinks(workspace: Workspace, view: View, url: String): String {
```
Comment (jp7677) [inline src/main/kotlin/nl/avisi/structurizr/site/generatr/site/DiagramGenerator.kt:40]: nit: free line above?

File: src/main/kotlin/nl/avisi/structurizr/site/generatr/site/DiagramGenerator.kt [hunk 2]
```diff
@@ -34,22 +36,18 @@ fun generateDiagrams(workspace: Workspace, exportDir: File) {
...
     val diagram = generatePlantUMLDiagramWithElementLinks(workspace, view, url)
 
     val name = "${diagram.key}-${view.key}"
-    val plantUMLFile = File(pumlDir, "$name.puml")
-    if (!plantUMLFile.exists() || plantUMLFile.readText() != diagram.definition) {
-        saveAsSvg(diagram, svgDir, name)
-        saveAsPUML(diagram, plantUMLFile)
```
Comment (jp7677) [inline src/main/kotlin/nl/avisi/structurizr/site/generatr/site/DiagramGenerator.kt:47]: Are you sure that we no longer need the `svg` and `puml` files on disk? I seem to remember that we use those files for the `svg` or `puml` download links in the footer of an embedded svg.

Comment (dirkgroot) [reply src/main/kotlin/nl/avisi/structurizr/site/generatr/site/DiagramGenerator.kt:47]: As far as I can tell, these files are really only temporary. I've also tested that the "regular" download links still work. The links in the footer link to the PUML and SVG files that are separately generated in the [generateDiagrams function](https://github.com/avisi-cloud/structurizr-site-generatr/blob/main/src/main/kotlin/nl/avisi/structurizr/site/generatr/site/DiagramGenerator.kt#L16).

Comment (jp7677) [reply src/main/kotlin/nl/avisi/structurizr/site/generatr/site/DiagramGenerator.kt:47]: ah yeah, i should have looked at the complete file and not just the diff, thanks for clarifying!

**Pull request merged**